### PR TITLE
HW7

### PR DESCRIPTION
**Pull request by DataJoshua**
…of a project when a task is created

### Summary

A brief description of the pull request.

### How it works

Screenshots/screencasts of the pull request introduced functionality.

### Test plan

List of steps to manually test introduced functionality:

- Log in
- Go to http://localhost:3000

### Deploy notes

Notes regarding deployment the contained body of work.
These should note any db migrations, services, scripts, etc.

### Checklist:

- [ ] I have performed a self-review of my own code
- [ ] My pull request is shorter than 500 lines

### CODE REVIEW

File: app/controllers/tasks_controller.rb [hunk 1]
```diff
@@ -17,10 +17,10 @@ def new
   end
 
   def create
-    @task = Task.new(task_params.merge({ project: @project }))
+    @task = create_task.task
     authorize! @task
```
Comment (RasimKhusaenov) [inline app/controllers/tasks_controller.rb:21]: What if user not authorized?

File: app/interactors/comments/destroy.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    def call
+      context.fail!(error: "Comment was not deleted") unless destroy_comment
+    end
+
+    private
+
+    def destroy_comment
+      @destroy_comment = comment.destroy
+    end
```
Comment (RasimKhusaenov) [inline app/interactors/comments/destroy.rb:15]: Why you need method for it? And why you memoize result?

File: app/interactors/tasks/create.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module Tasks
+  class Create
+    include Interactor::Organizer
+
+    organize Tasks::Save,
+             Tasks::Create::SendNotification,
+             Tasks::Create::NotifyAllMembers
```
Comment (RasimKhusaenov) [inline app/interactors/tasks/create.rb:7]: ?
```suggestion
             Tasks::Create::SendNotifications
```

File: app/interactors/tasks/create/send_notification.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+module Tasks
+  class Create
+    class SendNotification
+      include Interactor
+
+      delegate :task, :user, to: :context
+
+      def call
+        context.users = task.project.users
+        context.creator = user
```
Comment (RasimKhusaenov) [inline app/interactors/tasks/create/send_notification.rb:10]: Why you need them in context?

File: app/interactors/tasks/create.rb (hunk 3 above)
Comment (RasimKhusaenov) [reply app/interactors/tasks/create.rb:7]: Why you send mails two times?

File: app/interactors/tasks/delete.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+module Tasks
+  class Delete
```
Comment (RasimKhusaenov) [inline app/interactors/tasks/delete.rb:2]: Name it `Destroy`

File: app/interactors/tasks/delete.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+    include Interactor::Organizer
+
+    organize Tasks::Destroy,
```
Comment (RasimKhusaenov) [inline app/interactors/tasks/delete.rb:5]:
```suggestion
    organize Tasks::Destroy::Execute,
```

File: app/interactors/tasks/edit.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+module Tasks
+  class Edit
```
Comment (RasimKhusaenov) [inline app/interactors/tasks/edit.rb:2]: `Update`

File: app/interactors/tasks/edit.rb [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+    include Interactor::Organizer
+
+    organize Tasks::Update,
```
Comment (RasimKhusaenov) [inline app/interactors/tasks/edit.rb:5]:
```suggestion
    organize Tasks::Update::Execute,
```

File: app/mailers/task_mailer.rb [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    @members.each do |member|
+      @message = if ownership(member).role == :owner && member == user
+                   "You deleted your task, named '#{@task.title}'"
+                 elsif member == user
+                   "You deleted a task, named '#{@task.title}'"
+                 elsif ownership(member).role == :owner
+                   "A employee of you deleted a task, named '#{@task.title}'"
+                 else
+                   "Someone of you deleted a task, named '#{@task.title}'"
+                 end
```
Comment (RasimKhusaenov) [inline app/mailers/task_mailer.rb:24]: Interesting idea 👍

File: app/mailers/task_mailer.rb [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    @task = task
+    @creator = creator
+    mail(to: user.email)
+  end
+
+  # mailers for delete
+  def task_deleted(task, user, project)
+    @members = project.users
+    @task = task
+    @message = ""
```
Comment (RasimKhusaenov) [inline app/mailers/task_mailer.rb:13]: Why?

File: config/database.yml [hunk 11]
```diff
@@ -22,8 +22,13 @@ default: &default
...
 development:
-  <<: *default
-  database: task_tracker_development
+  adapter: postgresql
+  encoding: unicode
+  database: MyDatabase
+  host: localhost
+  pool: 5
+  username: postgres
+  password: MyPassword
```
Comment (RasimKhusaenov) [inline config/database.yml:31]: Don't change this file

**Review by RasimKhusaenov — COMMENTED** [review]
Where is interactors and mails for Project? :)

**Comment by DataJoshua** [comment]
Interactors for project are in process )

**Pull request closed without merging**